app/src/main/java/org/p2p/wallet/infrastructure/transactionmanager/impl/TransactionWorker.kt

```diff
@@ -0,0 +1,101 @@
+package org.p2p.wallet.infrastructure.transactionmanager.impl
+
+import android.content.Context
+import androidx.work.CoroutineWorker
+import androidx.work.WorkerParameters
+import org.koin.core.component.KoinComponent
+import org.p2p.wallet.infrastructure.transactionmanager.repository.TransactionQueueRepository
+import org.p2p.wallet.notification.AppNotificationManager
+import org.p2p.wallet.notification.ErrorTransactionNotification
+import org.p2p.wallet.notification.SwapTransactionNotification
+import org.p2p.wallet.rpc.repository.history.RpcHistoryRepository
+import org.p2p.wallet.transaction.model.TransactionExecutionState
+import timber.log.Timber
+
+class TransactionWorker(
+    context: Context,
+    workerParams: WorkerParameters,
+    private val rpcRepository: RpcHistoryRepository,
+    private val appNotificationManager: AppNotificationManager,
+    private val transactionQueueRepository: TransactionQueueRepository
+) : CoroutineWorker(context, workerParams), KoinComponent {
+
+    private val transactionId = inputData.getString(KEY_TRANSACTION_ID).orEmpty()
+    private val transaction = transactionQueueRepository.getTransactionById(transactionId)
+
+    companion object {
+        private const val TAG = "Transaction Worker"
+        const val KEY_TRANSACTION_ID = "TRANSACTION_ID"
+    }
+
+    init {
+        Timber.tag(TAG)
+            .d("New worker created: ${transaction?.serializedTransaction}")
+    }
+
+    override suspend fun doWork(): Result {
+        if (transaction == null) {
+            return Result.failure()
+        }
+        val serializedTransaction = transaction.serializedTransaction
+        val isSimulation = transaction.isSimulation
+        Timber
+            .tag(TAG)
+            .d("Transaction worker started: $serializedTransaction")
+
+        try {
```

Comment:
i think it can be rewritten using runCatching that returns `Result`

Reply:
i did not refactor code, i just moved implementation from old version, if we need it, i can refactor, but it's need testing

Reply:
yeah, okay, understood